modules/kernel/mkosi.clangd

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+set -eu
+
+if [ -z "${MKOSI_CONFIG-}" ]; then
+    exec mkosi \
```

Comment:
We shouldn't assume this script is invoked from the `mkosi-kernel` source directory. So this should use the `BASH_SOURCE` magic to figure out the directory this script is located in and then pass that to `--directory` with `../..` to make sure we run mkosi in the right directory

Reply:
No we indeed shouldn't and that's also why I resolve the path to the build script below. However, I would assume it's more likely that the user wants mkosi to scan the config from the current dir than from `mkosi-kernel` (e.g. when including the latter)?

Reply:
I think that's the more unlikely scenario.

The more likely scenario is that a user has a separate kernel tree open in their editor and wants to run clangd from mkosi-kernel within that editor workspace. And that means $PWD will not be the mkosi-kernel repository and instead we should chdir() to it with --directory

Reply:
In that case they can simply create a dummy `mkosi.conf` with an include statement for mkosi-kernel or prepend `env --chdir ...` to the invocation of the clangd wrapper script. If we always chdir to the mkosi-kernel directory we would make it completely impossible to use mkosi-kernel with `Include=` whereas the current state makes that trivial all while your scenario is still quite simple